rama-http/src/headers/x_robots_tag/robots_tag.rs

```diff
@@ -0,0 +1,102 @@
+use crate::headers::util::value_string::HeaderValueString;
+use crate::headers::x_robots_tag::custom_rule::CustomRule;
+use crate::headers::x_robots_tag::max_image_preview_setting::MaxImagePreviewSetting;
+use crate::headers::x_robots_tag::robots_tag_builder::RobotsTagBuilder;
+
+macro_rules! getter_setter {
+    ($field:ident, $type:ty) => {
+        paste::paste! {
+            pub fn [<$field>](&self) -> $type {
+                self.[<$field>]
+            }
+
+            pub fn [<set_ $field>](&mut self, [<$field>]: $type) -> &mut Self {
+                self.[<$field>] = [<$field>];
+                self
+            }
+        }
+    };
+
+    ($field:ident, $type:ty, optional) => {
+        paste::paste! {
+            pub fn [<$field>](&self) -> Option<&$type> {
+                self.[<$field>].as_ref()
+            }
+
+            pub fn [<set_ $field>](&mut self, [<$field>]: $type) -> &mut Self {
+                self.[<$field>] = Some([<$field>]);
+                self
+            }
+        }
+    };
+
+    ($field:ident, $type:ty, vec) => {
```

Comment:
I don't think you need this, as pretty sure you only need this ever for `custom_rules`, so you can take it out of the macro_rules